test/artifacts-helper/scenario_executable_wrapper.sh

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env bash
+
+set -e
+
+WRAPPER_INSTALL_PATH='/usr/local/share/codespace-features'
+BINS_TO_CHECK=(npm npx pnpm pnpx rush-pnpm rush yarn)
+
+check_path_priority() {
+    echo "Checking PATH priority"
+
+    for BIN_NAME in "${BINS_TO_CHECK[@]}"; do
+        if ! command -v "$BIN_NAME" &>/dev/null; then
+            echo "Error: $BIN_NAME not found in PATH"
+            exit 1
+        fi
+
+        # Check if the target binary is wrapped
+        ACTUAL_BIN_PATH=$(command -v "$BIN_NAME")
+        EXPECTED_BIN_PATH="$WRAPPER_INSTALL_PATH/$BIN_NAME"
+        if ! grep -q "$EXPECTED_BIN_PATH" <<<"$ACTUAL_BIN_PATH"; then
+            echo "Error: $BIN_NAME is not wrapped. We expected $EXPECTED_BIN_PATH but the actual binary path is $ACTUAL_BIN_PATH."
+            echo "Please contact Clipchamp EngProd and let them know that the feed-auth-wrapper feature is not working as expected."
+            exit 1
+        fi
+
+        echo "Success: $BIN_NAME is wrapped."
+    done
+}
+
+check_bin_exec() {
+    echo "Checking if the wrapped binaries get executed"
+
+    cat <<EOF>"$HOME/ado-auth-helper"
+#!/usr/bin/env bash
+
+echo "dummy-token"
+EOF
+    chmod +x "$HOME/ado-auth-helper"
+
+    WRAPPED_BINS_DIR=$(mktemp -d)
+    BASH_BIN_DIR=$(dirname "$(command -v bash)")
+    TEST_PATH="$WRAPPER_INSTALL_PATH:$WRAPPER_INSTALL_PATH:$WRAPPED_BINS_DIR:$BASH_BIN_DIR"
```

Comment:
I don't think this actually makes the executable wrapper appear twice in `which -a`. Probably best to copy or symlink the wrappers to simulate that scenario. It's important to prevent infinite recursion for the wrapper and this would make for a pretty robust test in that regard.